google/generativeai/types/model_types.py

```diff
@@ -183,16 +186,71 @@ class TuningExampleDict(TypedDict):
     output: str
 
 
-TuningExampleOptions = Union[TuningExampleDict, glm.TuningExample, tuple[str, str]]
+TuningExampleOptions = Union[TuningExampleDict, glm.TuningExample, tuple[str, str], list[str]]
 TuningDataOptions = Union[
-    glm.Dataset, Iterable[TuningExampleOptions]
-]  # TODO(markdaoust): csv, json, pandas, np
+    pathlib.Path, str, glm.Dataset, Mapping[str, Iterable[str]], Iterable[TuningExampleOptions]
+]  # TODO(markdaoust): json-URLS, gs:// URLS .. use etils.epath?
 
 
-def encode_tuning_data(data: TuningDataOptions) -> glm.Dataset:
+def encode_tuning_data(
+    data: TuningDataOptions, input_key="text_input", output_key="output"
+) -> glm.Dataset:
+    if isinstance(data, str):
+        # Strings are either URLs or system paths.
+        if re.match("^\w+://\S+$", data):
+            data = _normalize_url(data)
+        else:
+            # Normalize system paths to use pathlib
+            data = pathlib.Path(data)
+
+    if isinstance(data, pathlib.Path):
+        if data.suffix.lower() == ".json":
+            with open(data) as f:
+                data = json.load(f)
+
+    if isinstance(data, (str, pathlib.Path)):
+        import pandas as pd
+
+        data = pd.read_csv(data)
+
     if isinstance(data, glm.Dataset):
         return data
+    elif hasattr(data, "keys"):
+        return _convert_dict(data, input_key, output_key)
+    else:
+        return _convert_iterable(data)
+
+
+def _normalize_url(url: str) -> str:
+    sheet_base = "https://docs.google.com/spreadsheets"
```

Comment:
I don't think it will work for the parsing logic here, but  `gspread` can return `sheets.googleapis.com` URLs (see cl/526209504).

No action, just an FYI in case it shows up here too.